### DISCUSSION TITLE
what URL format that mssql accepts to connect a sqlserver?

### DISCUSSION THREAD

**Discussion by yadongyaly** [microsoft/sqltoolsservice#2372] [Q&A]
I am working on TypeScript project, trying to use windows.open(URL, _self) to connect a database with VSC mssql extension, but I cannot find a correct URL format that mssql extension accepts.

For example, I tried this:
const URL = vscode://ms-mssql.mssql//connect?server=${server}&database=${database}&authenticationType=AzureMFA;
window.open(URL, "_self");

After running, the vscode app pops out a windows saying "do you want to open this URL with mssql extension, and I clicked yes. However, nothing happens after that.

There is an example of opening Azure Data Studio, it works well, I am looking for something similar with VSC mssql extension.
const URL = azuredatastudio://openConnectionDialog?server=${server}&database=${database}&authenticationType=AzureMFA;

**Comment by Charles-Gagnon** [comment]
The URI handlers are implemented in the extension side, so don't have anything to do directly with SqlToolsService.

See responses here for status on support in the VS Code MSSQL extension : https://github.com/microsoft/vscode-mssql/issues/17915
